tests/macros/InterfaceOverview.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Test the API Interface Overview page content macro
+ *
+ * @prettier
+ */
+
+// Get necessary modules
+const fs = require('fs');
+const path = require('path');
+
+const { assert, describeMacro, beforeEachMacro, itMacro } = require('./utils');
+const jsdom = require('jsdom');
+const { JSDOM } = jsdom;
+
+const INTERFACE_DATA = `[${JSON.stringify({
+    XMLHttpRequestEventTarget: {
+        inh: 'EventTarget',
+        impl: []
+    }
+})}]`;
+
+const DATA_DIR = path.resolve(__dirname, 'fixtures', 'interface-overview');
+const DATA = ((...dataNames) => {
+    /** @type {Record<string, object>} */
+    let result = {};
+    for (let i = 0; i < dataNames.length; i++) {
+        const name = dataNames[i];
+        result[name] = JSON.parse(
+            fs.readFileSync(path.resolve(DATA_DIR, `${name}.json`))
+        );
+    }
+    return result;
+})('EventTarget', 'XMLHttpRequestEventTarget');
+
+/**
+ * @param {string[]} headers
+ * @returns {(result: string) => void}
+ */
+function testInterface(headers) {
+    return result => {
```

Comment:
I don't like that this is a function that returns a function because down below where you use it, it looks like a code error.

I'd prefer to see code below that was like `.then(result => checkSectionTitles(result, [...])`

I think that would make it a lot easier to understand.